apps/aeoracle/src/aeo_query_tx.erl

```diff
@@ -117,12 +117,16 @@ origin(#oracle_query_tx{sender = SenderPubKey}) ->
 check(#oracle_query_tx{sender = SenderPubKey, nonce = Nonce,
                        oracle = OraclePubKey, query_fee = QFee,
                        query_ttl = QTTL, response_ttl = RTTL,
-                       fee = Fee} = QTx, _Context, Trees, Height, _ConsensusVersion) ->
+                       fee = Fee} = QTx, Context, Trees, Height, _ConsensusVersion) ->
     Checks =
         [fun() -> aetx_utils:check_account(SenderPubKey, Trees, Nonce, Fee + QFee) end,
-         fun() -> aeo_utils:check_ttl_fee(Height, QTTL, Fee - ?ORACLE_QUERY_TX_FEE) end,
-         fun() -> check_oracle(OraclePubKey, Trees, QFee, Height, QTTL, RTTL) end,
+         fun() -> check_oracle(OraclePubKey, Trees, QFee, Height, OTTL, RTTL) end,
          fun() -> check_query(QTx, Trees, Height) end
+         | case Context of
+               aetx_contract -> [];
+               _ ->
```

Comment:
I would suggest to specify this clause with `aetx_transaction` so to prevent errors to slip through in case of refactoring or typos.